src/internal/connector/sharepoint/collection.go

```diff
@@ -217,126 +221,117 @@ func (sc *Collection) retrieveLists(
 	ctx context.Context,
 	wtr *kw.JsonSerializationWriter,
 	progress chan<- struct{},
+	errs *fault.Errors,
 ) (numMetrics, error) {
-	var (
-		errs    error
-		metrics numMetrics
-	)
+	var metrics numMetrics
 
-	lists, err := loadSiteLists(ctx, sc.service, sc.fullPath.ResourceOwner(), sc.jobs)
+	lists, err := loadSiteLists(ctx, sc.service, sc.fullPath.ResourceOwner(), sc.jobs, errs)
 	if err != nil {
-		return metrics, errors.Wrap(err, sc.fullPath.ResourceOwner())
+		return metrics, err
 	}
 
 	metrics.attempts += len(lists)
 	// For each models.Listable, object is serialized and the metrics are collected.
 	// The progress is objected via the passed in channel.
 	for _, lst := range lists {
+		if errs.Err() != nil {
+			break
+		}
+
 		byteArray, err := serializeContent(wtr, lst)
 		if err != nil {
-			errs = support.WrapAndAppend(*lst.GetId(), err, errs)
-			if sc.ctrl.FailFast {
-				return metrics, errs
-			}
-
+			errs.Add(clues.Wrap(err, "serializing list").WithClues(ctx))
 			continue
 		}
 
-		arrayLength := int64(len(byteArray))
+		size := int64(len(byteArray))
 
-		if arrayLength > 0 {
+		if size > 0 {
 			t := time.Now()
 			if t1 := lst.GetLastModifiedDateTime(); t1 != nil {
 				t = *t1
 			}
 
-			metrics.totalBytes += arrayLength
+			metrics.totalBytes += size
 
 			metrics.success++
 			sc.data <- &Item{
 				id:      *lst.GetId(),
 				data:    io.NopCloser(bytes.NewReader(byteArray)),
-				info:    sharePointListInfo(lst, arrayLength),
+				info:    sharePointListInfo(lst, size),
 				modTime: t,
 			}
 
 			progress <- struct{}{}
 		}
 	}
 
-	return metrics, nil
+	return metrics, errs.Err()
 }
 
 func (sc *Collection) retrievePages(
 	ctx context.Context,
 	wtr *kw.JsonSerializationWriter,
 	progress chan<- struct{},
+	errs *fault.Errors,
 ) (numMetrics, error) {
-	var (
-		errs    error
-		metrics numMetrics
-	)
+	var metrics numMetrics
 
 	betaService := sc.betaService
 	if betaService == nil {
-		return metrics, fmt.Errorf("beta service not found in collection")
+		return metrics, clues.New("beta service required").WithClues(ctx)
 	}
 
 	pages, err := sapi.GetSitePages(ctx, betaService, sc.fullPath.ResourceOwner(), sc.jobs)
 	if err != nil {
-		return metrics, errors.Wrap(err, sc.fullPath.ResourceOwner())
+		return metrics, err
```

Comment:
Should we retain info about `sc.fullPath.ResourceOwner()`?

Reply:
That data should already be present in the context.